### ISSUE TITLE
Disk quota exceeded: OCI runtime error

### ISSUE THREAD

**Issue by DavidMGau** [containers/podman-desktop#8452]
### Bug description

After creating and destroying a number of containers, I can suddenly no longer create containers. The run command fails with

`Error: crun: create keyring `410742b2d2de473d2cb955f29b6988303489c911db754cfeb48c80478866d843`: Disk quota exceeded: OCI runtime error`

and no amount of deletion, pruning and restarting makes any difference.

There are two workarounds.
1. Delete and recreate the machine (losing all volume data in the process)
2. SSH into the pod host
    `podman machine ssh <your-machine-name>`
    `sudo su`
    `echo "400" > /proc/sys/kernel/keys/maxkeys`

So it seems that there is a small, fixed limit of distinct containers that can be produced, after which the user must start again or make a surgical change to the pod machine OS.

Can we have a pod machine default image that does not have this limitation?

### Operating system

macOS

### Installation Method

Brew (macOS)

### Version

1.12.0

### Steps to reproduce

Start some containers.
Recreate them a few times.
Eventually, any new run command will fail with a "disk quota exceeded" error.

### Relevant log output

```shell
Error: crun: create keyring `410742b2d2de473d2cb955f29b6988303489c911db754cfeb48c80478866d843`: Disk quota exceeded: OCI runtime error
Error: crun: create keyring `89ae8855cc67a39daef7740da79c8023a0524dd10e9bc00fdcf8fc58162a4ae0`: Disk quota exceeded: OCI runtime error
Error: crun: create keyring `c394894c3ab0f0d27342c03ec297a6e5ed5b56a82a0b2051c0f9dda1572fb875`: Disk quota exceeded: OCI runtime error
Error: crun: create keyring `783f60db71e1e6109fc1d7c97d1c0a8a89a277f7c44702f68c73bb4b5cbd9561`: Disk quota exceeded: OCI runtime error
Error: crun: create keyring `5423df58c235aaf3b7f9e4c75973058e20f686db66b79ff0fdcd651368f0bbbb`: Disk quota exceeded: OCI runtime error
Error: crun: create keyring `4374266bab813b8d371e445ef76357dc7582c73eabc2c718dbfb90b1ead278eb`: Disk quota exceeded: OCI runtime error
Error: crun: create keyring `8e397839b65724d84ab07ffd7cafb4d07a73d79b0c420769f50723469ba67a84`: Disk quota exceeded: OCI runtime error
```

**Comment by ghost** [comment]
I have also stumbled upon this. Original bug report against podman https://github.com/containers/podman/issues/13363.  Since on Windows and macOS, this is happening with VM images under `podman-machine` control, I think that it would make sense to set more sensible defaults for the image OS. @giuseppe

**Comment by usma0118** [comment]
+1 also facing same on Mac.

**Comment by odockal** [comment]
@DavidMGau @aaccioly-zepz @usma0118 Thanks for opening the report folks! Can you please open an issue on podman side since this is upstream issue for us, not something podman desktop can do about.

**Comment by ajram23** [comment]
+1

**Comment by odockal** [comment]
The issue was opened on the podman side some days ago: https://github.com/containers/podman/issues/23784.

**Comment by ajram23** [comment]
Yup posted my comment there as well. For now doing this took care of it. "podman machine ssh
sudo sysctl -w kernel.keys.maxkeys=20000" I don't get the reason why it keeps track of the count. Especially when the images are deleted.

**Comment by odockal** [comment]
@ajram23 Thanks for adding the workaround. The fix seems to be in gvproxy 0.7.5 that is part of (afaik) podman 5.2.3 release. We will include that in 1.13.0.

**Comment by ajram23** [comment]
@odockal Thank you! But I moved away from Podman to a simpler solution. Much appreciate the fix though!

**Comment by benoitf** [comment]
closing as we already include the newer gvproxy version since a long time
